File: Sources/Server/fairy-teller/src/main/java/com/rad/fairyteller/service/PurchaseService.java
package com.rad.fairyteller.service;

import com.rad.fairyteller.domain.Purchase;
import com.rad.fairyteller.repository.PurchaseRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
@RequiredArgsConstructor
public class PurchaseService {

    private final PurchaseRepository purchaseRepository;

    public void saveOrUpdatePurchase(final Purchase purchase) {
        Optional.ofNullable(purchase).ifPresent(purchaseRepository::save);
    }

    public List<Purchase> getPurchaseListByUserId(final Long userId) {
        return purchaseRepository.findAllByUserId(userId);
    }

    public Purchase getById(final Long id) {
        return purchaseRepository.findOne(id);
    }
}

File: Sources/Server/fairy-teller/src/main/java/com/rad/fairyteller/service/PurchaseService.kt
package com.rad.fairyteller.service

import com.rad.fairyteller.domain.purchase.Purchase
import com.rad.fairyteller.repository.PurchaseRepository
import org.springframework.stereotype.Service

@Service
class PurchaseService(private val purchaseRepository: PurchaseRepository) {

    fun saveOrUpdatePurchase(purchase: Purchase) {
        purchaseRepository.save(purchase)
    }

    fun getPurchaseListByUserId(userId: Long?): List<Purchase> {
        return purchaseRepository.findAllByUserId(userId)
    }

    fun getById(id: Long?): Purchase {
        return purchaseRepository.findOne(id)
    }
}